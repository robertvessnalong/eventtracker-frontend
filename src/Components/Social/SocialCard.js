import { useState, useEffect } from 'react';
import { Card } from 'react-bootstrap';
import { IoAirplane } from 'react-icons/io5';
import FlightHubApi from '../../API/api';

const SocialCard = ({ comment }) => {
  return (
    <Card className='Favortie mb-4'>
      <Card.Header>
        {' '}
        {comment.departureAirport ?? 'N/A'}
        <IoAirplane /> {comment.arrivalAirport ?? 'N/A'}
      </Card.Header>
      <Card.Body>
        <Card.Title className='h3'>{comment.comment}</Card.Title>
        <Card.Text>
          - {comment.user.firstName} {comment.user.lastName}
        </Card.Text>
      </Card.Body>
      <Card.Footer className='text-muted'>
        {FlightHubApi.FlightTimeConverter(comment.createdAt)}
      </Card.Footer>
    </Card>
  );
};

export default SocialCard;
