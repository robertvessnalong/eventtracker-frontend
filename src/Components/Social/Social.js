import { useState, useEffect } from 'react';
import { Container, Col, Row, Card, Spinner } from 'react-bootstrap';
import FlightHubApi from '../../API/api';
import SocialCard from './SocialCard';
import uuid from 'react-uuid';

const Social = () => {
  const [comments, setComments] = useState([]);
  const [loading, setLoading] = useState(true);

  // useEffect(() => {
  //   const getComments = async () => {
  //     const res = await FlightHubApi.getComments();
  //     setLoading(false);
  //     setComments(res);
  //   };
  //   getComments();
  // }, []);

  // if (loading) {
  //   return (
  //     <Row>
  //       <Col style={{ textAlign: 'center' }}>
  //         <Spinner animation='border' role='status'>
  //           <span className='visually-hidden'>Loading...</span>
  //         </Spinner>
  //       </Col>
  //     </Row>
  //   );
  // }

  return (
    <div className='Social'>
      {/* <header>
        <h1>Social</h1>
        <span>See what everyone is saying about their flights!</span>
      </header>
      <hr></hr>
      <Container>
        <Row>
          {comments.map((comment, idx) =>
            idx % 2 === 0 ? (
              <Col key={uuid()} md={{ span: 8, offset: 0 }}>
                <SocialCard comment={comment} />
              </Col>
            ) : (
              <Col key={uuid()} md={{ span: 8, offset: 4 }}>
                <SocialCard comment={comment} />
              </Col>
            )
          )}
        </Row>
      </Container> */}
    </div>
  );
};

export default Social;
