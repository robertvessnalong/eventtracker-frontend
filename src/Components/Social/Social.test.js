import {
  render,
  screen,
  waitForElementToBeRemoved,
} from '@testing-library/react';
import '@testing-library/jest-dom';
import Social from './Social';
import TestUser from '../../Testing/TestUser';
import UserContext from '../../Context/UserContext';
import { BrowserRouter } from 'react-router-dom';

describe('Social Component', () => {
  // Smoke Test
  test('renders without crashing', async () => {
    render(
      <BrowserRouter>
        <UserContext.Provider value={{ user: TestUser }}>
          <Social />
        </UserContext.Provider>
      </BrowserRouter>
    );
    await waitForElementToBeRemoved(screen.getByText('Loading...'));
    const label = screen.getByText(/I am ready for this event!/i);
    expect(label).toBeInTheDocument();
  });

  // Snapshot
  test('matches snapshot', async () => {
    const { asFragment } = render(
      <BrowserRouter>
        <UserContext.Provider value={{ user: TestUser }}>
          <Social />
        </UserContext.Provider>
      </BrowserRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
