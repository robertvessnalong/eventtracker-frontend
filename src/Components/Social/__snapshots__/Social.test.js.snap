// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <div
      class="col"
      style="text-align: center;"
    >
      <div
        class="spinner-border"
        role="status"
      >
        <span
          class="visually-hidden"
        >
          Loading...
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
